.devcontainer/script.sh

```diff
@@ -23,4 +23,16 @@ if [ -z "$ARM_SUBSCRIPTION_ID" ]; then
     exit 0
 fi
 
-az login --service-principal -u $ARM_CLIENT_ID -p $ARM_CLIENT_SECRET --tenant $ARM_TENANT_ID
\ No newline at end of file
+# If ARM_CLIENT_CERTIFICATE is set, decode it and save to a temp file
+if [ -n "$ARM_CLIENT_CERTIFICATE" ]; then
+    echo "$ARM_CLIENT_CERTIFICATE" | base64 -d > /tmp/certfile.pem
+    export ARM_CLIENT_CERTIFICATE_PATH="/tmp/certfile.pfx"
+    export ARM_CLIENT_CERTIFICATE_PASSWORD=$(date '+%s')
+    echo "$ARM_CLIENT_CERTIFICATE" | base64 -d | openssl pkcs12 -export -password pass:"${ARM_CLIENT_CERTIFICATE_PASSWORD}" -out $ARM_CLIENT_CERTIFICATE_PATH
```

Comment:
is this echo necessary?

Reply:
yes the echo gives the input to the pipe